src/ObjCRuntime/Runtime.cs

```diff
@@ -575,9 +575,6 @@ static unsafe IntPtr GetMethodFromToken (uint token_ref)
 
 		static unsafe IntPtr GetGenericMethodFromToken (IntPtr obj, uint token_ref)
 		{
-#if MONOMAC
-			throw new NotSupportedException ();
```

Comment:
if that's the case then maybe `#if MONOMAC && !XAMCORE_2_0` would be more appropriate